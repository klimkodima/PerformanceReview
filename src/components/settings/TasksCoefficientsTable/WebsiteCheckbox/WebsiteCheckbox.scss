.custom-checkbox {
  &.MuiCheckbox-root {
    padding: 0;
  }
  color: #919191;
  svg {
    fill: #919191;
  }

  &.Mui-checked {
    color: #919191;
    svg {
      fill: #a4a4a4;
    }
  }

  &__check-icon {
    position: absolute;
  }
}
