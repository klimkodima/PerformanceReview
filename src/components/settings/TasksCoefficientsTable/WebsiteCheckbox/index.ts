export { default as CustomCheckbox } from './WebsiteCheckbox';
