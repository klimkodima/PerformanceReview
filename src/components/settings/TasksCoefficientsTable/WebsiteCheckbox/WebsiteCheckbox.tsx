import { ChangeEvent, memo, ReactElement } from 'react';

import { Checkbox } from '@mui/material';
import CheckIcon from '@mui/icons-material/Check';
import CheckBoxOutlineBlankIcon from '@mui/icons-material/CheckBoxOutlineBlank';

import './WebsiteCheckbox.scss';

type CustomCheckboxPropsType = {
  checked: boolean;
  onChange: (event: ChangeEvent<HTMLInputElement>) => void;
};

const WebsiteCheckbox = memo(
  ({ checked, onChange }: CustomCheckboxPropsType): ReactElement => (
    <Checkbox
      className='custom-checkbox'
      data-testid='custom-checkbox'
      checked={checked}
      onChange={onChange}
      checkedIcon={
        <>
          <CheckBoxOutlineBlankIcon />
          <CheckIcon className='custom-checkbox__check-icon' />
        </>
      }
    />
  )
);

export default WebsiteCheckbox;

WebsiteCheckbox.displayName = 'WebsiteCheckbox';
