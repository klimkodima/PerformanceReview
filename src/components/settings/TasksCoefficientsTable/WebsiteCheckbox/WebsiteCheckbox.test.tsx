import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import WebsiteCheckbox from './WebsiteCheckbox';

describe('WebsiteCheckbox', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(<WebsiteCheckbox checked={false} onChange={() => true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('show check icon', () => {
    render(<WebsiteCheckbox checked={true} onChange={() => true} />);
    expect(screen.getByTestId('CheckIcon')).toBeInTheDocument();
  });
});
