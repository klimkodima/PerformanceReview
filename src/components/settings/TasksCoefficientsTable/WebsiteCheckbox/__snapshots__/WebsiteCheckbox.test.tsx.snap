// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebsiteCheckbox renders correctly 1`] = `
<span
  className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
  data-testid="custom-checkbox"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={null}
>
  <input
    checked={false}
    className="PrivateSwitchBase-input css-1m9pwf3"
    data-indeterminate={false}
    onChange={[Function]}
    type="checkbox"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="CheckBoxOutlineBlankIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
    />
  </svg>
</span>
`;
